impeller/renderer/backend/vulkan/texture_source_vk.h

```diff
@@ -23,10 +28,26 @@ class TextureSourceVK {
 
   virtual vk::ImageView GetImageView() const = 0;
 
-  bool SetLayout(const BarrierVK& barrier) const;
+  /// Encodes the layout transition `barrier` to `barrier.cmd_buffer` for the
+  /// image.
+  ///
+  /// The transition is from the layout stored via `SetLayoutWithoutEncoding` to
+  /// `barrier.new_layout`.
+  fml::Status SetLayout(const BarrierVK& barrier) const;
```

Comment:
Yea, this one was tricky since does "true" mean a transition happened or does "true" mean no error happened.  I figured it's cleaner to just use the proper type.  I wish we used it more often.